src/metrics/kms-service.ts

```diff
@@ -19,8 +24,9 @@ export class KMSService {
         result = await kmsClient.decrypt({ CiphertextBlob: buffer }).promise();
       } catch {
         // Then try with encryption context, in case API key was encrypted using the AWS Console
-        const encryptionContext = { LambdaFunctionName: process.env.AWS_LAMBDA_FUNCTION_NAME ?? "" };
-        result = await kmsClient.decrypt({ CiphertextBlob: buffer, EncryptionContext: encryptionContext }).promise();
+        result = await kmsClient
+          .decrypt({ CiphertextBlob: buffer, EncryptionContext: this.encryptionContext })
+          .promise();
```

Comment:
Same for this? I wonder if this code somehow would fail and throw something from our side?

Reply:
It always could throw, we catch and log here, where we don't re-raise https://github.com/DataDog/datadog-lambda-js/blob/main/src/metrics/listener.ts#L185

Reply:
I missed that, thank you!